chain/src/chain.rs

```diff
@@ -863,17 +869,21 @@ impl Chain {
 
 		let header = self.get_block_header(&h)?;
 
-		{
-			let mut txhashset_ref = self.txhashset.write();
-			// Drop file handles in underlying txhashset
-			txhashset_ref.release_backend_files();
-		}
-
-		// Rewrite hashset
-		txhashset::zip_write(self.db_root.clone(), txhashset_data, &header)?;
-
-		let mut txhashset =
-			txhashset::TxHashSet::open(self.db_root.clone(), self.store.clone(), Some(&header))?;
+		// Write hashset to sandbox
+		txhashset::clean_txhashset_sandbox(self.db_root.clone());
+		txhashset::zip_write(
+			self.db_root.clone(),
```

Comment:
Can we more explicitly separate the `sandbox` from the actual `txhashset` dir?
I don't think these should share the same `db_root` - its going to be easy for someone to unintentionally change this code in the future without realizing they are now writing to the _real_ txhashset dir.

Reply:
accept.
changed to `env::temp()` folder for sandbox.